node/external/logs/logsRepository.go

```diff
@@ -0,0 +1,62 @@
+package logs
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+type logsRepository struct {
+	storer      storage.Storer
+	marshalizer marshal.Marshalizer
+}
+
+func newLogsRepository(args argsNewLogsRepository) *logsRepository {
+	storer := args.StorageService.GetStorer(dataRetriever.TxLogsUnit)
+
+	return &logsRepository{
+		storer:      storer,
+		marshalizer: args.Marshalizer,
+	}
+}
+
+func (repository *logsRepository) getLog(logKey []byte, epoch uint32) (*transaction.Log, error) {
+	bytes, err := repository.storer.GetFromEpoch(logKey, epoch)
+	if err != nil {
+		return nil, err
+	}
+
+	txLog := &transaction.Log{}
+	err = repository.marshalizer.Unmarshal(txLog, bytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return txLog, nil
+}
+
+// Since a lookup table of logs is useful in the "logsFacade" to JOIN "txs" with "logs" ON "tx.hash" == "log.key",
+// we'll return such a structure here (the returned logs don't have to be in any particular order).
+func (repository *logsRepository) getLogs(logsKeys [][]byte, epoch uint32) (map[string]*transaction.Log, error) {
+	keyValuePairs, err := repository.storer.GetBulkFromEpoch(logsKeys, epoch)
+	if err != nil {
+		return nil, err
+	}
+
+	results := make(map[string]*transaction.Log, len(logsKeys))
```

Comment:
can remove the `len(logsKeys)` part

Reply:
Removed. Shouldn't we add the capacity hint, though, in some cases (when handling a large number of items)?

Reply:
for maps it doesn't matter. We mainly use that for slices.